src/fineman/elimination_algorithm.py

```diff
@@ -0,0 +1,91 @@
+from math import ceil
+
+from fineman.betweenness_reduction import betweenness_reduction
+from fineman.elimination_by_hop_reduction import elimination_of_r_remote_edges_by_hop_reduction
+from fineman.helper_functions import reweight_graph, super_source_bfd, compute_reach, transpose_graph, b_hop_sssp, \
+    b_hop_stsp
+from fineman.independent_set_or_crust import find_is_or_crust
+
+
+def _eliminate_independent_set(graph, independent_set):
+    out_I = {(u,v) for u in independent_set for v in graph[u].keys()}
+    graph_out_I, _ = _subgraph_with_out_set(graph, out_I)
+    return super_source_bfd(graph_out_I, out_I, 1)
+
+
+def _compute_price_functiom_to_make_U_r_remote(graph, neg_edges, negative_sandwich, beta):
```

Comment:
```suggestion
def _compute_price_function_to_make_U_r_remote(graph, neg_edges, negative_sandwich, beta):
```

Reply:
Proposition for later changes: Maybe we should consider just letting the function actually make the set U r-remote.